consensus/broadcast/metaChainMessenger_test.go

```diff
@@ -303,3 +306,96 @@ func TestMetaChainMessenger_BroadcastBlockDataLeader(t *testing.T) {
 		assert.Equal(t, len(transactions), numBroadcast)
 	})
 }
+
+func TestMetaChainMessenger_Close(t *testing.T) {
+	args := createDefaultMetaChainArgs()
+	varModified := false
+	delayedBroadcaster := &mock.DelayedBroadcasterMock{
+		CloseCalled: func() {
+			varModified = true
+		},
+	}
+	args.DelayedBroadcaster = delayedBroadcaster
+	mcm, _ := broadcast.NewMetaChainMessenger(args)
+	if mcm == nil {
+		return
+	}
+	mcm.Close()
+	assert.True(t, varModified)
+}
+
+func TestMetaChainMessenger_PrepareBroadcastHeaderValidator(t *testing.T) {
+	t.Run("Nil header", func(t *testing.T) {
+		args := createDefaultMetaChainArgs()
+		checkVarModified := false
+		delayedBroadcaster := &mock.DelayedBroadcasterMock{
+			SetHeaderForValidatorCalled: func(vData *shared.ValidatorHeaderBroadcastData) error {
+				checkVarModified = true
+				return nil
+			},
+		}
+		args.DelayedBroadcaster = delayedBroadcaster
+		mcm, _ := broadcast.NewMetaChainMessenger(args)
+		if mcm == nil {
+			return
+		}
+		mcm.PrepareBroadcastHeaderValidator(nil, make(map[uint32][]byte), make(map[string][][]byte), 0, make([]byte, 0))
+		assert.False(t, checkVarModified)
+	})
+	t.Run("Err on core.CalculateHash", func(t *testing.T) {
+		args := createDefaultMetaChainArgs()
+		checkVarModified := false
+		delayedBroadcaster := &mock.DelayedBroadcasterMock{
+			SetHeaderForValidatorCalled: func(vData *shared.ValidatorHeaderBroadcastData) error {
+				checkVarModified = true
```

Comment:
maybe do a require.Fail(t, "SetHeaderForValidator should not be called") instead of the check for False at line 359